CIP-0185? | Governance Action Addendums (#1180)
cardano-foundation/CIPs: CIP-0185/README.md

---
CIP: 185
Title: Governance Action Addendums
Category: Metadata
Status: Proposed
Authors:
    - Nicolas Cerny <nicolas.cerny@cardanofoundation.org>
    - Pi Lanningham <pi@sundaeswap.finance>
Implementors: []

> Discussions:
>     - https://github.com/cardano-foundation/CIPs/pull/1180
>     - https://github.com/cardano-foundation/CIPs/pull/1182

Reviewer: ```suggestion
Discussions:
    - Original PR: https://github.com/cardano-foundation/CIPs/pull/1180
    - Concurrent PR (closed): https://github.com/cardano-foundation/CIPs/pull/1182
```
This will fix the validation problem currently reported here: https://github.com/cardano-foundation/CIPs/actions/runs/31160001114/job/92807938789?pr=1180

- **64-byte limit compliance:** chunking the `uri` into an array ensures transactions are accepted by the ledger, which rejects metadata strings over 64 bytes, and that indexers can reliably reconstruct the pointer.
- **Always-display philosophy:** an earlier draft required indexers to discard updates that failed validation or arrived after a terminal state. This revision follows CIP-0100's guidance to surface content with prominent warnings instead, because silent suppression risks censoring legitimate voices via indexer bugs and hides impersonation attempts that are themselves relevant signal.
- **Publishing updates as new governance actions** was rejected: it is heavyweight, requires deposits and re-voting, and does not address the post-ratification case where an administrator needs a clarification of an action that has already passed.

> - **Relationship to CIP-0184:** this proposal was drafted concurrently with [CIP-0184][] (Governance Proposal Feedback and Addenda), and this revision incorporates several of its ideas — the update taxonomy, the author-complete signing rule, the no-widening constraints, structured `bindingStatements`, and the explicit bootstrapping of authority — reflected in its co-authorship. This CIP deliberately retains a narrower scope: a single document type covering author-issued updates to *submitted* governance actions, with a fully-specified on-chain anchoring and indexing flow. Pre-submission draft proposals and third-party feedback documents (CIP-0184's `DraftProposal` and `ProposalFeedback`) are complementary concerns and remain out of scope here.

Reviewer: ```suggestion
- **Relationship to CIP-0184 candidate:** this proposal was drafted concurrently with [CIP-0184?][] (Governance Proposal Feedback and Addenda), and this revision incorporates several of its ideas — the update taxonomy, the author-complete signing rule, the no-widening constraints, structured `bindingStatements`, and the explicit bootstrapping of authority — reflected in its co-authorship. This CIP deliberately retains a narrower scope: a single document type covering author-issued updates to *submitted* governance actions, with a fully-specified on-chain anchoring and indexing flow. Pre-submission draft proposals and third-party feedback documents (the CIP-0184 candidate's `DraftProposal` and `ProposalFeedback`) are complementary concerns and remain out of scope here.
```
Unless https://github.com/cardano-foundation/CIPs/pull/1182 is reopened at @Quantumplation's discretion, there is not going to be a CIP-0184.  There was not one at the time of that number assignment either: only when merged into the repository does that become a CIP.  So to be accurate both now & in the future we need some changes to the wording.

Since the CIP editors sometimes have to retract widely publicised statements about drafts and other incomplete works being called "CIPs" as soon as they are posted — which defeats the purpose of a review process — please also try to pass along this request to others who are standardising their work & try to imagine the chaos there would be if drafts were considered CIPs (i.e. standards) as soon as they were posted as a pull request. 🙏

(note the shortcut reference link will also have to be changed below: https://github.com/cardano-foundation/CIPs/pull/1180#discussion_r3739375534)

Reviewer 2: (apologies for not getting to a deeper review yet, but)

I do think this policy is harmful... ontologically, as soon as they exist in someone's head, they are CIP's: cardano improvement proposals. They are someones Proposal to Improve Cardano. Just like an RFC is an RFC as soon as it is posted: it is a Request For Comments, whether it actually gets widely adopted or not.

What you're objecting to is more that people imbue them with *authority* before the review process; An ideal process would have a separate term for those that had gone through review and been merged, perhaps Cardano Standard.

That being said, we live in an imperfect world, so I doubt we're likely to upend the entrenched process and terminology based purely on that. Just know that I'm over here grumpily grumbling.

Reviewer: thanks @Quantumplation - we've been consistently encouraging the community to continue with those proposed improvements as "candidates" (i.e. the ideation & initial expression) since that term carries the same enthusiasm without declaring finality.

After some problems in the past, I think this taking better hold in the community thanks to repetition in the Community Digest which focuses on what it means to be a "candidate" CIP and what is accomplished when a submission is "completed"; e.g. [this recent column](https://forum.cardano.org/t/digest-july-22-2026-van-rossem-hard-fork-activates-constitutional-committee-election-cardano-foundation-updates-cardano-improvement-proposals-cip-updates-spotlight-on-learn-cardano-linda/155842#p-386040-cardano-improvement-proposals-cip-updates-6).

Reviewer 2: *grumble grumble ATM machine grumble* lol

## Path to Active

### Acceptance Criteria

- [ ] The JSON-LD context and schema for `GovernanceActionUpdate` are finalized, with published test vectors (canonicalized forms and blake2b-256 hashes) alongside the example documents.
- [ ] At least one metadata creation tool supports generating, signing, and anchoring documents conforming to this schema.
- [ ] At least one major governance interface (e.g. GovTool, Adastat, Cexplorer, 1694.io) implements detection, witness verification, and rendering of the append-only update log, including author-completeness labelling.

### Implementation Plan

- Solicit feedback from governance tooling providers on the detection and indexing flow for label `1694` update anchors.
- Provide test vectors and reference documents for wallet, CLI, and UI developers.
- Coordinate with at least one governance explorer to prototype the update-log display and use the prototype to pressure-test the schema.

## Copyright

This CIP is licensed under [CC-BY-4.0][].

> [CIP-0010]: ../CIP-0010/README.md
> [CIP-0100]: ../CIP-0100/README.md
> [CIP-0108]: ../CIP-0108/README.md

Reviewer: ```suggestion
[CIP-0010]: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0010/README.md
[CIP-0100]: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0100/README.md
[CIP-0108]: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0108/README.md
```
Relative links for CIPs have never been portable... they're used but they've been hit-or-miss and so authors are generally advised to use absolute URLs: i.e. GitHub URLs because their immutability makes them canonical.

See [CIP-1852](https://github.com/cardano-foundation/CIPs/blob/master/CIP-1852/README.md) to show how relative links in the GitHub source document are inadequate: the `../` works properly on GitHub (follow the relative link to CIP-0003 to confirm) but the different framework on `cips.cardano.org` makes that link fail on [its version of CIP-1852](https://cips.cardano.org/cip/CIP-1852).

Nobody has pressed for a relative linking format that works in both contexts & I believe I added something to CIP-0001 in years gone by that advises everyone to use absolute links.  Relative links were [added in the meantime](https://github.com/cardano-foundation/cf-cip-frontend/issues/51) but only to "assets" in the same CIP/CPS.

If you believe a universal relative linking format is necessary — which would refer to a _different_ CIP or CPS _in the same document archive_ (i.e. without your `cips.cardano.org` link opening on GitHub) — then you can leave this text alone <ins>but then</ins>:
* We would have to open an issue on [cf-cip-frontend](https://github.com/cardano-foundation/cf-cip-frontend) to ensure relative inter-CIP links are supported, hopefully by the time this is merged (cc @aldo555) which would also fix all the currently broken relative inter-CIP links in other CIPs (cc @Ryun1).

> [CIP-0184]: https://github.com/cardano-foundation/CIPs/pull/1182

Reviewer: ```suggestion
[CIP-0184?]: https://github.com/cardano-foundation/CIPs/pull/1182
```
(continued from previous https://github.com/cardano-foundation/CIPs/pull/1180#discussion_r3739373625: to show CIP candidate tag correctly)
